Tests/MapboxMapsTests/MapView/Unit Tests/MapInitOptionsTests.swift

```diff
@@ -0,0 +1,142 @@
+
+        let mapView = MapView(frame: .zero, mapInitOptions: mapInitOptions)
+        let resourceOptions = try! mapView.__map.getResourceOptions()
+
+        XCTAssertEqual(resourceOptions, mapInitOptions.resourceOptions)
+        XCTAssertEqual(resourceOptions.accessToken, credentialsManager.accessToken)
+    }
+
+    func testOptionsAreSetFromNibDataSource() {
+        let credentialsManager = CredentialsManager(accessToken: "pk.dddddd")
+
+        dataSourceReturnValue = MapInitOptions(
+            resourceOptions: ResourceOptions(accessToken: credentialsManager.accessToken))
+
+        // Load view from a nib, where the map view's datasource is the file's owner,
+        // i.e. this test.
+        let testBundle = Bundle.module
+        let nib = UINib(nibName: "MapInitOptionsTests", bundle: testBundle)
+
+        // Instantiate the view.
+        let objects = nib.instantiate(withOwner: self, options: nil)
+
+        let view = objects.first { object -> Bool in
+            return (object as? MapView)?.tag == 1
+        }
+
+        guard let mapView = view as? MapView else {
+            XCTFail("Not a MapView")
+            return
+        }
+        XCTAssertNotNil(mapView.mapInitOptionsDataSource)
+
+        guard let optionsFromDataSource = mapView.mapInitOptionsDataSource?.mapInitOptions() as? MapInitOptions else {
+            XCTFail("MapInitOptions not returned from data source")
+            return
+        }
+
+        guard let dataSourceReturnValue = self.dataSourceReturnValue as? MapInitOptions else {
+            XCTFail("dataSourceReturnValue not a MapInitOptions")
+            return
+        }
+
+        // Check that the dataSource in MapView is correctly wired, so that the
+        // expected options are returned
+        XCTAssertEqual(optionsFromDataSource, dataSourceReturnValue)
+
+        // Now check the resource options from the initialized MapView
+        let resourceOptions = try! mapView.__map.getResourceOptions()
+
+        XCTAssertEqual(resourceOptions, dataSourceReturnValue.resourceOptions)
+        XCTAssertEqual(resourceOptions.accessToken, credentialsManager.accessToken)
+    }
+
+    func testDefaultOptionsAreUsedWhenNibDoesntSetDataSource() {
+        XCTAssert(dataSourceReturnValue is Bool )
+        CredentialsManager.default.accessToken = "pk.eeeeee"
+
+        // Load view from a nib, where the map view's datasource is nil
+        let testBundle = Bundle.module
+        let nib = UINib(nibName: "MapInitOptionsTests", bundle: testBundle)
+
+        // Instantiate the view.
+        let objects = nib.instantiate(withOwner: self, options: nil)
+
+        // Second view doesn't have data source connected.
+        let view = objects.first { object -> Bool in
+            return (object as? MapView)?.tag == 2
+        }
+
+        guard let mapView = view as? MapView else {
+            XCTFail("Not a MapView")
+            return
+        }
```

Comment:
I think this could be simpler given that we control the `xib`…

```suggestion
        let mapView = objects.compactMap { $0 as? MapView }.first { $0.tag == 2 }!
```